### PR TITLE
docs - materialized views

### PR DESCRIPTION
**Pull request by mkiyama**
--Added/updated commands
--Added/updated catalog views
--Added/updated Admin. Guide topics

dev PR https://github.com/greenplum-db/gpdb/pull/8662

link to HTML page with description of changes and links to topics
http://docs-msk-gpdb6-dev.cfapps.io/6-0/ref_guide/0-mat-view.html

Link to individual topics in HTML format

**Admin Guide Changes**
http://docs-msk-gpdb6-dev.cfapps.io/6-0/admin_guide/ddl/ddl-mat-view.html
http://docs-msk-gpdb6-dev.cfapps.io/6-0/admin_guide/dml.html

**New commands**
http://docs-msk-gpdb6-dev.cfapps.io/6-0/ref_guide/sql_commands/ALTER_MATERIALIZED_VIEW.html
http://docs-msk-gpdb6-dev.cfapps.io/6-0/ref_guide/sql_commands/CREATE_MATERIALIZED_VIEW.html
http://docs-msk-gpdb6-dev.cfapps.io/6-0/ref_guide/sql_commands/DROP_MATERIALIZED_VIEW.html
http://docs-msk-gpdb6-dev.cfapps.io/6-0/ref_guide/sql_commands/REFRESH_MATERIALIZED_VIEW.html

**Changed commands**
http://docs-msk-gpdb6-dev.cfapps.io/6-0/ref_guide/sql_commands/ALTER_TABLE.html
http://docs-msk-gpdb6-dev.cfapps.io/6-0/ref_guide/sql_commands/COMMENT.html
http://docs-msk-gpdb6-dev.cfapps.io/6-0/ref_guide/sql_commands/CREATE_INDEX.html
http://docs-msk-gpdb6-dev.cfapps.io/6-0/ref_guide/sql_command/CREATE_MATERIALIZED_VIEW.html
http://docs-msk-gpdb6-dev.cfapps.io/6-0/ref_guide/sql_commands/CREATE_VIEW.html

**New/Changed catalog views**
http://docs-msk-gpdb6-dev.cfapps.io/6-0/ref_guide/system_catalogs/pg_matviews.html
http://docs-msk-gpdb6-dev.cfapps.io/6-0/ref_guide/system_catalogs/pg_rewrite.html
http://docs-msk-gpdb6-dev.cfapps.io/6-0/ref_guide/system_catalogs/pg_rules.html

### CODE REVIEW

**Comment by mkiyama** [comment]
@gaos1, @vraghavan78 
Do I have the correct reviewers for the materialized view doc updates?

thanks

File: gpdb-doc/dita/admin_guide/ddl/ddl-mat-view.xml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    <body>
+      <p>The <codeph>REFRESH MATERIALIZED VIEW</codeph> command updates the materialized view
+        data.</p>
+      <p>
+        <codeblock>REFRESH MATERIALIZED VIEW us_users;</codeblock>
+      </p>
+      <p>With the <codeph>WITH NO DATA</codeph> clause, the current data is removed, no new data is
+        generated, and the materialized view is left in an unscannable state. An error is returned
+        if a query attempts to access an unscannable materialized view.</p>
+      <codeblock>REFRESH MATERIALIZED VIEW us_users WITH NO NEW DATA;</codeblock>
```
Comment (gaos1) [inline gpdb-doc/dita/admin_guide/ddl/ddl-mat-view.xml]: should be WITH NO DATA

File: gpdb-doc/dita/ref_guide/system_catalogs/pg_rules.xml [hunk 2]
```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE topic
+  PUBLIC "-//OASIS//DTD DITA Composite//EN" "ditabase.dtd">
+<topic id="topic1" xml:lang="en">
+  <title id="gi143896">pg_rules</title>
+  <body>
+    <p>The view <codeph>pg_matviews</codeph> provides access to useful information about query
```
Comment (litzell) [inline gpdb-doc/dita/ref_guide/system_catalogs/pg_rules.xml]: pg_matviews -> pg_rules?

File: gpdb-doc/dita/admin_guide/ddl/ddl-mat-view.xml (hunk 1 above)
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/ddl/ddl-mat-view.xml]: fixed

File: gpdb-doc/dita/admin_guide/ddl/ddl-mat-view.xml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  <topic xml:lang="en" id="topic_hn3_xy5_kjb">
+    <title>Creating Materialized Views</title>
+    <body>
+      <p>The <codeph>CREATE MATERIALIZED VIEW </codeph>command defines a materialized view based on
+        a query.</p>
+      <p>
+        <codeblock>CREATE MATERIALIZED VIEW us_users AS SELECT u.id, u.name, a.zone FROM users u, address a WHERE a.country = 'USA';</codeblock>
+      </p>
+      <p>If a materialized view query contains an <codeph>ORDER BY</codeph> or <codeph>SORT</codeph>
+        clause, the clause is ignored when the a <codeph>SELECT</codeph> is performed on the
```
[the remark this answers is not in the document]
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/ddl/ddl-mat-view.xml]: fixed

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_MATERIALIZED_VIEW.xml [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+      <parml>
+        <plentry>
+          <pt><varname>table_name</varname></pt>
+          <pd>The name (optionally schema-qualified) of the materialized view to be created.</pd>
+        </plentry>
+        <plentry>
+          <pt><varname>column_name</varname></pt>
+          <pd>The name of a column in the materialized view. The column names are assigned based on
+            position. The first column name is assigned to the first column of the query result, and
+            so on. If a column name is not provided, it are taken from the output column names of
```
[the remark this answers is not in the document]
Comment (mkiyama) [reply gpdb-doc/dita/ref_guide/sql_commands/CREATE_MATERIALIZED_VIEW.xml]: fixed

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_MATERIALIZED_VIEW.xml [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+        at the time of view creation. Though <codeph>*</codeph> was used to create the materialized
+        view, columns added later to the table will not be part of the view. </p>
+      <p>Create a view that gets the top ten ranked baby names:</p>
+      <codeblock>CREATE MATERIALIZED VIEW topten AS SELECT name, rank, gender, year FROM 
+names, rank WHERE rank &lt; '11' AND names.id=rank.id;</codeblock>
+    </section>
+    <section id="section7">
+      <title>Compatibility</title>
+      <p><codeph>CREATE MATERIALIZED VIEW</codeph> is a Greenplum Database extension of the SQL
+        standard..</p>
```
[the remark this answers is not in the document]
Comment (mkiyama) [reply gpdb-doc/dita/ref_guide/sql_commands/CREATE_MATERIALIZED_VIEW.xml]: fixed

File: gpdb-doc/dita/ref_guide/sql_commands/REFRESH_MATERIALIZED_VIEW.xml [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+        <plentry>
+          <pt>CONCURRENTLY</pt>
+          <pd>Refresh the materialized view without locking out concurrent selects on the
+            materialized view. Without this option, a refresh that affects a lot of rows tends to
+            use fewer resources and completes more quickly, but could block other connections which
+            are trying to read from the materialized view. This option might be faster in cases
+            where a small number of rows are affected.<p>This option is only allowed if there is at
+              least one <codeph>UNIQUE</codeph> index on the materialized view which uses only
+              column names and includes all rows; that is, it must not index on any expressions nor
+              include a <codeph>WHERE</codeph> clause.</p><p>This option cannot not be used when the
```
[the remark this answers is not in the document]
Comment (mkiyama) [reply gpdb-doc/dita/ref_guide/sql_commands/REFRESH_MATERIALIZED_VIEW.xml]: fixed

File: gpdb-doc/dita/ref_guide/system_catalogs/pg_rules.xml (hunk 2 above)
Comment (mkiyama) [reply gpdb-doc/dita/ref_guide/system_catalogs/pg_rules.xml]: fixed

File: gpdb-doc/dita/admin_guide/ddl/ddl-mat-view.xml [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+<topic id="topic_i3c_qwy_sp">
+  <title id="im140342">Creating and Managing Materialized Views</title>
+  <body>
+    <p>Materialized views are similar to views. A materialized view enables you to save a frequently
+      used or complex query, then access the query results in a <codeph>SELECT</codeph> statement as
+      if they were a table. Materialized views persist the query results in a table-like form. While
+      access to the data stored in a materialized view can be much faster than accessing the
+      underlying tables directly or through a view, the data is not always current. </p>
+    <p>The materialized view data cannot be directly updated. To refresh the materialized view data,
+      use the <codeph>REFRESH MATERIALIZED VIEW</codeph> command. The query used to create the
```
[the remark this answers is not in the document]
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/ddl/ddl-mat-view.xml]: added example

File: gpdb-doc/dita/admin_guide/ddl/ddl-mat-view.xml [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    <p>The information about a materialized view in the Greenplum Database system catalogs is
+      exactly the same as it is for a table or view. A materialized view is a relation, just like a
+      table or a view. When a materialized view is referenced in a query, the data is returned
+      directly from the materialized view, just like from a table. The query in the materialized
+      view definition is only used for populating the materialized view.</p>
+    <p>If you can tolerate periodic updates of materialized view data, the performance benefit can
+      be substantial.</p>
+    <p>One use of a materialized view is to allow faster access to data brought in from an external
+      data source such as external table or a foreign data wrapper. Also, an index can be defined on
+      a materialized view, foreign data wrappers do not support indexes; this advantage might not
```
[the remark this answers is not in the document]
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/ddl/ddl-mat-view.xml]: fixed

File: gpdb-doc/dita/admin_guide/ddl/ddl-mat-view.xml [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+      <p>
+        <codeblock>CREATE MATERIALIZED VIEW us_users AS SELECT u.id, u.name, a.zone FROM users u, address a WHERE a.country = 'USA';</codeblock>
+      </p>
+      <p>If a materialized view query contains an <codeph>ORDER BY</codeph> or <codeph>SORT</codeph>
+        clause, the clause is ignored when a <codeph>SELECT</codeph> is performed on the
+        materialized query.</p>
+    </body>
+  </topic>
+  <topic xml:lang="en" id="topic_vwd_zy5_kjb">
+    <title>Refreshing Materialized Views</title>
```
[the remark this answers is not in the document]
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/ddl/ddl-mat-view.xml]: changed title

File: gpdb-doc/dita/ref_guide/sql_commands/REFRESH_MATERIALIZED_VIEW.xml [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+          <pt>CONCURRENTLY</pt>
+          <pd>Refresh the materialized view without locking out concurrent selects on the
+            materialized view. Without this option, a refresh that affects a lot of rows tends to
+            use fewer resources and completes more quickly, but could block other connections which
+            are trying to read from the materialized view. This option might be faster in cases
+            where a small number of rows are affected.<p>This option is only allowed if there is at
+              least one <codeph>UNIQUE</codeph> index on the materialized view which uses only
+              column names and includes all rows; that is, it must not index on any expressions nor
+              include a <codeph>WHERE</codeph> clause.</p><p>This option cannot be used when the
+              materialized view is not already populated and cannot be used with the <codeph>WITH NO
```
[the remark this answers is not in the document]
Comment (mkiyama) [reply gpdb-doc/dita/ref_guide/sql_commands/REFRESH_MATERIALIZED_VIEW.xml]: fixed

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_MATERIALIZED_VIEW.xml [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+    <section id="section3">
+      <title>Description</title>
+      <p><codeph>CREATE MATERIALIZED VIEW</codeph> defines a materialized view of a query. The query
+        is executed and used to populate the view at the time the command is issued (unless
+          <codeph>WITH NO DATA</codeph> is used) and can be refreshed using <codeph>REFRESH
+          MATERIALIZED VIEW</codeph>.</p>
+      <p><codeph>CREATE MATERIALIZED VIEW</codeph> is similar to <codeph>CREATE TABLE AS</codeph>,
+        except that it also remembers the query used to initialize the view, so that it can be
+        refreshed later upon demand. A materialized view has many of the same properties as a table,
+        but there is no support for temporary materialized views or automatic generation of
```
[the remark this answers is not in the document]
Comment (mkiyama) [reply gpdb-doc/dita/ref_guide/sql_commands/CREATE_MATERIALIZED_VIEW.xml]: added REFRESH MATERIALIZED VIEW information.

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_MATERIALIZED_VIEW.xml [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+    <p id="sql_command_desc">Defines a new materialized view.</p>
+    <section id="section1">
+      <title>Synopsis</title>
+      <codeblock id="sql_command_synopsis">CREATE MATERIALIZED VIEW <varname>table_name</varname>
+    [ (<varname>column_name</varname> [, ...] ) ]
+    [ WITH ( <varname>storage_parameter</varname> [= <varname>value</varname>] [, ... ] ) ]
+    [ TABLESPACE <varname>tablespace_name</varname> ]
+    AS <varname>query</varname>
+    [ WITH [ NO ] DATA ]
+    [DISTRIBUTED {| BY column [opclass], [ ... ] | RANDOMLY | REPLICATED }]</codeblock>
```
[the remark this answers is not in the document]
Comment (mkiyama) [reply gpdb-doc/dita/ref_guide/sql_commands/CREATE_MATERIALIZED_VIEW.xml]: fixed

**Comment by mkiyama** [comment]
@hlinnaka, @vraghavan78 
Any comments?

**Pull request merged**